frontend/explorer/public/js/hash.js

```diff
@@ -1209,25 +1209,100 @@ function populateHashPage(hash, explorerHash) {
+		appendNavigationMenuBlockstakeOutput(explorerHash.transactions, hash);
 		appendHeading(infoBody, 'Hash Type: BlockStake Output ID');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendBlockStakeOutputTables(infoBody, hash, explorerHash);
 	}
 }
 
+// appendNavigationMenuTranscaction adds the transaction link to the top navigation menu
+function appendNavigationMenuTransaction(explorerTransaction) {
+	appendNavigationMenuBlock(explorerTransaction);
+	var navigation = document.getElementById('nav-links');
+	var transactionSpan = document.createElement('span');
+	navigation.appendChild(document.createTextNode(' > '));
+	navigation.appendChild(transactionSpan);
+	linkHash(transactionSpan, explorerTransaction.id, 'Transaction');
+}
+
+// appendNavigationMenuCoinOuput adds the coin output link to the top navigation menu
+function appendNavigationMenuCoinOutput(explorerHash, hash) {
+	//Coin Ouput
+	if (explorerHash.transactions != null) {
+		for (var i = 0; i < explorerHash.transactions.length; i++) {
+			for (var j = 0; j < explorerHash.transactions[i].coinoutputids.length; j++) {
+				if (explorerHash.transactions[i].coinoutputids[j] == hash) {
+					appendNavigationMenuTransaction(explorerHash.transactions[i]);
+					var navigation = document.getElementById('nav-links');
+					var outputSpan = document.createElement('span');
+					navigation.appendChild(document.createTextNode(' > '));
+					navigation.appendChild(outputSpan);
+					linkHash(outputSpan, explorerHash.transactions[i].coinoutputids[j], 'Coin Output');
+				}
+			} 
+		}
+	//Coin Ouput - Block Creator Reward
+	} else if (explorerHash.blocks != null) {
+		for (var i = 0; i < explorerHash.blocks.length; i++) {
+			for (var j = 0; j < explorerHash.blocks[i].minerpayoutids.length; j++) {
+				if (explorerHash.blocks[i].minerpayoutids[j] == hash) {
+					appendNavigationMenuBlock(explorerHash.blocks[i]);
+					var navigation = document.getElementById('nav-links');
+					var outputSpan = document.createElement('span');
+					navigation.appendChild(document.createTextNode(' > '));
+					navigation.appendChild(outputSpan);
+					linkHash(outputSpan, explorerHash.blocks[i].minerpayoutids[j], 'Payout ID');
+				}
+			}
+		}
+	}
+}
+
+// appendNavigationMenuBlockstakeOuput adds the blockstake output link to the top navigation menu
+function appendNavigationMenuBlockstakeOutput(explorerTransactions, hash) {
+	for (var i = 0; i < explorerTransactions.length; i++) {
+		for (var j = 0; j < explorerTransactions[i].blockstakeoutputids.length; j++) {
+			if (explorerTransactions[i].blockstakeoutputids[j] == hash) {
+				appendNavigationMenuTransaction(explorerTransactions[i]);
+				var navigation = document.getElementById('nav-links');
+				var outputSpan = document.createElement('span');
+				navigation.appendChild(document.createTextNode(' > '));
+				navigation.appendChild(outputSpan);
+				linkHash(outputSpan, explorerTransactions[i].blockstakeoutputids[j], 'Blockstake Output');
```

Comment:
return when you're finished, so you do not keep looping when you're already finished.

Reply:
Added return statement